broadcast/src/linked/mod.rs

```diff
@@ -0,0 +1,663 @@
+//!   This enables reliable recovery and prevents conflicts like duplicate messages.
+//!
+//! # Design
+//!
+//! The system has two sets of nodes: Sequencers and Validators. The sets may overlap.
+//! Sequencers link messages, while Validators verify and sign them. Each message includes a
+//! threshold signature over its predecessor for an unbroken, verifiable chain.
+//!
+//! # Actor Overview and Design
+//!
+//! The core of the protocol is the `Signer` actor. It manages the lifecycle of a secure,
+//! ordered broadcast by:
+//!
+//! 1. **Initializing** storage, networking, and cryptographic settings.
+//! 2. **Processing** incoming messages by validating data and signatures.
+//! 3. **Broadcasting** signed messages to all nodes.
+//! 4. **Verifying** message integrity and order.
+//! 5. **Updating Epochs** based on consensus.
+//!
+//! ## Key Design Principles
+//!
+//! - **Actor-Based Model**:  
+//!   The `Signer` (see `actor.rs`) encapsulates all functions, handling network I/O,
+//!   verification, timeouts, and state updates concurrently.
+//!
+//! - **Event Sourcing & Journaling**:  
+//!   Each new message is logged to a persistent journal. This maintains chain integrity and
+//!   enables reliable recovery after crashes.
+//!
+//! - **Asynchronous Processing**:  
+//!   Using Rust’s async/await, the actor processes tasks simultaneously, including epoch
+//!   refreshes, message rebroadcasts, and verification requests.
+//!
+//! ## Operational Flow
+//!
+//! 1. **Message Chaining**:  
+//!    Each new message includes a threshold signature over the previous message, forming an
+//!    immutable, verifiable chain.
+//!
+//! 2. **Signature Aggregation**:  
+//!    Validators produce partial signatures that combine into a threshold signature,
+//!    ensuring quorum endorsement.
+//!
+//! 3. **Epoch Management**:  
+//!    Epochs define the active set of signers and sequencers. The actor refreshes its epoch and
+//!    enforces strict bounds on accepted messages.
+//!
+//! 4. **Reliable Recovery**:  
+//!    Storage logs all sequencer messages, allowing nodes to replay messages after shutdown.
+//!
+//! 5. **Concurrent Operations**:  
+//!    The actor listens for network events, dispatches verification tasks, and manages timeouts
+//!    for efficient operation.
+//!
+//! # Public Modules
+//!
+//! - **`signer`**: Contains the actor that validates messages, links them, and handles threshold
+//!   signatures.
+//! - **`prover`**: Generates and verifies "proofs" (i.e. threshold signatures over messages).
+//!
+//! # Internal Modules
+//!
+//! - **`namespace`**: Generates namespace IDs for different signature types.
+//! - **`serializer`**: Serializes and deserializes messages.
+//! - **`wire`**: Contains the protocol buffer-generated message definitions.
+//! - **`safe`**: Provides parsed wrappers around wire types.
+//! - **`mocks`**: Provides mock implementations for testing.
```

Comment:
We should acknowledge `Autobahn` at the end here imo.

One example of doing this is: https://github.com/commonwarexyz/monorepo/blob/9f67390d2757dbc7a9f7ddddafd8a660e1dfbb9d/cryptography/src/bls12381/primitives/mod.rs#L3-L10